undefined reference to 'log10'

Apologies if this is a dumb question. I can't quite tell if this is an error on my end or `llvm-bazel`. Any help appreciated.

When I build `@llvm-bazel//...` I'm seeing:

```
bazel-out/host/bin/external/llvm-project/llvm/_objs/Support/Signals.o:Signals.cpp:function printSymbolizedStackTrace(llvm::StringRef, void**, int, llvm::raw_ostream&): error: undefined reference to 'log10'
bazel-out/host/bin/external/llvm-project/llvm/_objs/Support/Signals.o:Signals.cpp:function printSymbolizedStackTrace(llvm::StringRef, void**, int, llvm::raw_ostream&): error: undefined reference to 'log10'
```

This is when linking `mlir/mlir-linalg-ods-yaml-gen` but it appears to happen as soon as it attempts to link anything. It seems there might be an `-lm` missing somewhere.

My configuration is essentially borrowed from `xls` https://github.com/google/xls/tree/main/dependency_support/llvm_bazel, but I played around with LLVM versions in case that made a difference.

```bzl
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

def llvm_workspace():
    # newest
    LLVM_COMMIT = "2f69975683f5b6ea7df79f335f96b889a4bddecd"

    # xls
    # LLVM_COMMIT = "628dda08b82fcedcd9e89c9ef7850388e988bf68"  # 2021-04-07
    # LLVM_SHA256 = "b48c8c63c17631cc0160e1359c1e977188aa6cf5924cfd4b5664397effe65f30"
    # LLVM_BAZEL_SHA256 = "259684aa19af62201bd6a261faf315aff8a17dc8bcafcaed372b14108a872f71"

    # llvm-bazel example
    # LLVM_COMMIT = "499bce3abab8a362b9b4197944bd40b826c736c4"
    # LLVM_SHA256 = "a154965dfeb2b5963acc2193bc334ce90b314acbe48430ba310d8a7c7a20de8b"
    # LLVM_BAZEL_SHA256 = "a05a83300b6b4d8b45c9ba48296c06217f3ea27ed06b7e698896b5a3b2ed498d"

    http_archive(
        name = "llvm-bazel",
        # sha256 = LLVM_BAZEL_SHA256,
        strip_prefix = "llvm-bazel-llvm-project-{llvm_commit}/llvm-bazel".format(llvm_commit = LLVM_COMMIT),
        url = "https://github.com/google/llvm-bazel/archive/refs/tags/llvm-project-{llvm_commit}.tar.gz".format(llvm_commit = LLVM_COMMIT),
    )

    http_archive(
        name = "llvm-project-raw",
        build_file_content = "#empty",
        # sha256 = LLVM_SHA256,
        strip_prefix = "llvm-project-" + LLVM_COMMIT,
        url = "https://github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
    )
```

and

```bzl
load("@llvm-bazel//:terminfo.bzl", "llvm_terminfo_disable")
load("@llvm-bazel//:zlib.bzl", "llvm_zlib_disable")
load("@llvm-bazel//:configure.bzl", "llvm_configure")

def llvm_initialize():
    llvm_terminfo_disable(
        name = "llvm_terminfo",
    )

    llvm_zlib_disable(
        name = "llvm_zlib",
    )

    # This macro creates a new repo containing the LLVM source code downloaded
    # as llvm-project-raw and the BUILD files from llvm-bazel (contained in the
    # directory llvm-project-overlay). This new repo is constructed as symlinks
    # into these two directories.
    llvm_configure(
        # Name of resulting repo (e.g., "@llvm-project")
        name = "llvm-project",

        # The path to the directory containing the LLVM BUILD files within
        # llvm-bazel.
        overlay_path = "llvm-project-overlay",
        overlay_workspace = "@llvm-bazel//:WORKSPACE",

        # Project containing the llvm source code downloaded with http_archive.
        src_path = ".",
        src_workspace = "@llvm-project-raw//:WORKSPACE",
    )
```

Really appreciate this effort. Thanks for your help.

--- a/llvm-bazel/llvm-project-overlay/llvm/BUILD
+++ b/llvm-bazel/llvm-project-overlay/llvm/BUILD
@@ -215,6 +215,7 @@ cc_library(
         "//conditions:default": [
             "-pthread",
             "-ldl",
+            "-lm",
         ],
     }),
     textual_hdrs = glob([

--- a/llvm-bazel/llvm-project-overlay/mlir/BUILD
+++ b/llvm-bazel/llvm-project-overlay/mlir/BUILD
@@ -209,13 +209,6 @@ cc_library(
         "include/mlir/Pass/*.h",
     ]),
     includes = ["include"],
-    linkopts = select({
-        "@bazel_tools//src/conditions:windows": [],
-        "//conditions:default": [
-            "-lm",
-            "-lpthread",
-        ],
-    }),
     deps = [
         ":IR",
         ":Support",
@@ -4238,10 +4231,6 @@ cc_library(
 cc_binary(
     name = "mlir-cpu-runner",
     srcs = ["tools/mlir-cpu-runner/mlir-cpu-runner.cpp"],
-    linkopts = select({
-        "@bazel_tools//src/conditions:windows": [],
-        "//conditions:default": ["-ldl"],
-    }),
     deps = [
         ":AllToLLVMIRTranslations",
         ":ExecutionEngineUtils",
@@ -4373,13 +4362,6 @@ cc_binary(
         "tools/mlir-tblgen/*.h",
         "tools/mlir-tblgen/*.cpp",
     ]),
-    linkopts = select({
-        "@bazel_tools//src/conditions:windows": [],
-        "//conditions:default": [
-            "-lm",
-            "-lpthread",
-        ],
-    }),
     deps = [
         ":MlirTableGenMain",
         ":Support",
@@ -4395,13 +4377,6 @@ cc_binary(
     srcs = [
         "tools/mlir-linalg-ods-gen/mlir-linalg-ods-gen.cpp",
     ],
-    linkopts = select({
-        "@bazel_tools//src/conditions:windows": [],
-        "//conditions:default": [
-            "-lm",
-            "-lpthread",
-        ],
-    }),
     deps = [
         ":IR",
         ":Support",

